### ISSUE TITLE
ActiveRecord updates stalling

### ISSUE THREAD

**Issue by ticketevo** [goncalossilva/acts_as_paranoid#47]
tl;dr - We add acts as paranoid. Its doing soft deletes correctly, but updates with models with many associations are taking a long time or stalling.

We are working with a data model with many nested level associations -- a person has many dogs, dogs has many toys, toys has many properties, etc. Each of these models has acts_as_paranoid setup correctly on themselves.

When we have a person that has lots of dogs, who have lots of toys, etc, and try to update the name of a person and then save, the save stalls. Our logs show no SQL queries happening and nothing is getting destroyed.

We just wanted to ask, if anyone else had experienced a similar behavior? We are still investigating the problem, but wanted to start a dialog of our findings.

**Comment by goncalossilva** [comment]
The code has changed a lot since this report. Does it still stall on master?

**Comment by goncalossilva** [comment]
I'm going to close this for now. _Please_ re-open this if you can still reproduce this with the latest version.
